app/models/manageiq/providers/autosde/inventory/persister/storage_manager.rb

```diff
@@ -19,4 +19,14 @@ def initialize_inventory_collections
     add_collection(storage, :ext_management_system)
     add_collection(storage, :storage_service_resource_attachments)
   end
+
+  def physical_storage_details
```

Comment:
You're actually overwriting the accessor method `persister.physical_storage_details` that is being used by the parser so each time you call `persister.physical_storage_details.build` it is calling this method.

```suggestion
    add_physical_storage_details
  end

  def add_physical_storage_details
```